src/main/java/jenkins/plugins/logstash/LogstashConfiguration.java

```diff
@@ -45,9 +46,30 @@
   public LogstashConfiguration()
   {
     load();
+    if (enabled == null)
+    {
+      if (logstashIndexer == null)
+      {
+        enabled = false;
+      }
+      else
+      {
+        enabled = true;
+      }
+    }
     activeIndexer = logstashIndexer;
   }
 
+  public boolean isEnabled()
+  {
+    return enabled;
```

Comment:
potential NPE here, maybe use Optional<Boolean> instead?

Reply:
This should be covered by the constructor. It is checking after `load()` if enabled is set.

Reply:
I see, but this is still very easy to break. Can we add an explicit guard here if you don't want to use optional (`enabled == null ? false : enabled`)?

BTW strange that FB doesn't flag this as potential NPE

Reply:
Probably FB assumes due to Constructor that it can't be null.